node/cn/api_backend.go

```diff
@@ -76,8 +76,10 @@ func (b *CNAPIBackend) CurrentBlock() *types.Block {
 }
 
 func (b *CNAPIBackend) SetHead(number uint64) {
-	//b.cn.protocolManager.downloader.Cancel()
+	b.cn.protocolManager.Downloader().Cancel()
```

Comment:
How downloader start again after `SetHead`?

Reply:
After setHead logic is done, Td gets smaller than peer's. 
If local Td is smaller than peer's Td, then new goroutine starts downloader automatically.
```
if trueTD.Cmp(pm.blockchain.GetTd(currentBlock.Hash(), currentBlock.NumberU64())) > 0 {
	go pm.synchronise(p)
}
```